README.md

```diff
@@ -2,24 +2,49 @@
 
 The cloud credential operator is a controller that will sync on
 CredentialsRequest custom resources. CredentialsRequests allow OpenShift
-components to request fine grained credentials for a particular cloud provider.
+components to request fine-grained credentials for a particular cloud provider
 (as opposed to using the admin credentials, or elevated permissions granted via
-instance roles)
-
-# Design Principles
-
-  * The controller should be able to run in either a cluster itself, or in a centralized management cluster, most likely along side [Hive](https://github.com/openshift/hive).
+instance roles).
+
+- [Design Principles](#design-principles)
+- [Cloud Providers](#cloud-providers)
+  - [Credentials Root Secret Formats](#credentials-root-secret-formats)
+  - [AWS](#aws)
+  - [Azure](#azure)
+  - [GCP](#gcp)
+  - [Kubevirt](#kubevirt)
+  - [OpenStack](#openstack)
+  - [Ovirt](#ovirt)
+  - [VSphere:](#vsphere)
+- [Modes of Operation](#modes-of-operation)
+  - [1. Mint Mode](#1-mint-mode)
+    - [1.1 Mint Mode With Removal/Rotation Of Admin Credential](#11-mint-mode-with-removalrotation-of-admin-credential)
+  - [2. Passthrough Mode](#2-passthrough-mode)
+  - [3. Manual Credentials Management](#3-manual-credentials-management)
+  - [4. Short Lived Tokens](#4-short-lived-tokens)
+- [Support Matrix](#support-matrix)
+- [Developer Instructions](#developer-instructions)
+  - [Deploying in cluster](#deploying-in-cluster)
+- [Credentials Requests](#credentials-requests)
+  - [AWS Sample](#aws-sample)
+  - [Azure Sample](#azure-sample)
+- [Instructions to add new cloud provider](#instructions-to-add-new-cloud-provider)
+- [For OpenShift Second Level Operators](#for-openshift-second-level-operators)
+
+## Design Principles
+
+  * The controller should be able to run in either a cluster itself, or in a centralized management cluster, most likely alongside [Hive](https://github.com/openshift/hive).
```

Comment:
I do not any specific relation between Hive and CCO here. Do we need this in design principles?